### ISSUE TITLE
Can't save

### ISSUE THREAD

**Issue by bsabiston** [jasarien/Provenance#243]
Hello,

  I'm sorry to bring this up again -- I had the same issue a month ago and thought it was resolved.  I just cannot get certain game saves to work.  Specifically I'm trying to play Zelda Link to the Past, and it refuses to save.  I'm not using save states -- I am trying to use 'native' game saving. I turned off the auto saving of states in case that was fouling things up.
Last month I thought it was a bad rom issue, but I've downloaded Roms from several places and none of them work.  The current one I have is from the no-intro set.
Anyone else have this issue? It might just be with Zelda.  I don't have the same trouble with Chrono Trigger or Super Metroid.
Thanks
Bob

**Comment by jasarien** [comment]
I haven't been able to reproduce this issue. Anyone else?

**Comment by vintageluke** [comment]
I'm having the same issue. Playing BoF2 on snes, and the first time it deleted everything. Now it at least has the game save's original creation save, but won't save beyond that. Same thing happening with Lufia 2.

**Comment by vintageluke** [comment]
Update: Quitting a game from the menu and not just killing the app seems to keep the save. So far anyway.

**Comment by Jaysue21** [comment]
Hello, i just updated and now i don't have a save state menu as before, when i hit the menu button on my controller it goes straight back to rom screen, when i enter the game again it starts over fresh, i have auto save on, is there another way, thanks.

**Comment by jasarien** [comment]
@jaysue21 are you sure you have the very latest code? The issue with the menu button in controllers was fixed recently.

**Comment by Jaysue21** [comment]
I thought i was, i downloaded Providence from the iPwn store because it was having trouble with the top shelf update through xCode just kept getting a code signing error, the version i got was 1.2.6 thought that was the latest.

**Comment by jasarien** [comment]
Whatever version is on ipwn is very likely not the latest. I don't know who maintains that site, but it's not me.

**Comment by Jaysue21** [comment]
I know, it was just easier for me to install from there, i guess i can try again with xCode, cause that save thing is a pain.

**Comment by thedrick** [comment]
@jaysue21 battery saves (native saves) only get commited when you exit the game through the menu. If you exit by going to the Apple TV home screen it won't commit your save.

@jasarien I noticed battery saves only get commited when `stopEmulation` is called, which isn't called when the app just closes without dismissing the emulator view controller

**Comment by jasarien** [comment]
@thedrick thanks for the heads up. I'll have a look to see what can be done to more reliably commit the saves. That being said, I'm not sure that this is the cause of the original issue, unless people really are just quitting and not using the in-app menu.

**Comment by Jaysue21** [comment]
When I soft press menu on my Nimbus controller it goes straight to the rom screen and nothing get saved, when I select auto save and continue where I left off toggle I get a scrambled screen on some games. I emailed ipwn and they said they had the latest version 1.2.6, but I showed them the history page and they said they would look into it, so I'm assuming I don't have the last installed.

**Comment by thedrick** [comment]
@jaysue21 yeah you'll need to update to the latest code, that specific bug was just fixed a couple of days ago.

**Comment by Jaysue21** [comment]
@thedrick that's sounds right ipwn said they would fix it so I'm hoping for that unless I try with the Xcode again, had some trouble with that method, not very familiar with Xcode.

**Comment by lolkthxbai** [comment]
Hey guys, I too have the same issue as @jaysue21 
I'm running tvOS 9.2 and Provenance 1.2.6
Is there a changelog? I can't seem to find one
And is this the latest release as of now? Thanks!

**Comment by lolkthxbai** [comment]
@jasarien I checked the provenance info.plist on GitHub and it shows the version number is 1.2.6, which is exactly what I have now. The issue started after I updated my Apple TV to tvOS 9.2. The issue is continuous and I can't really tell what causes it. I will try to build the app again and see it resolves the issue.

**Comment by jasarien** [comment]
This was fixed in #311. Check out the latest code (the version number doesn't reflect the exact commit only major version changes).

**Comment by sevdestruct** [comment]
I was a bit confused about what is being reported here and if it was strictly a tvOS or also an iOS issue so I did some vigorous testing against this in iOS with Zelda. I can confirm and reproduce this issue and I think the name of this issue was throwing me off. The battery saves are only committed when returning to library - if you want to refer to that as quitting the game.. If you quit the app via app switcher, it saves nothing into battery save. I assume the same is happening in tvOS, but I can test that later too if needed.

**Comment by JoeMatt** [comment]
**Status update (2026):**

This is a long-standing confirmed bug — battery saves are held in memory and only flushed when cleanly returning to the library. A force-quit loses the last save interval.

This is still plausibly present. Investigation needed: does the app register a UIApplicationWillTerminateNotification / sceneDidDisconnect handler that forces a save flush? SwiftData migration (#2510, now in progress) may affect where/how saves are committed.

Tagging for investigation.

**Comment by JoeMatt** [comment]
@claude we do have code to automatically create save states on a crash, which is working well, we can hook in battery save shutdown calls here too.

noting the migration to swiftdata is interesting, not sure if the database being used will matter since battery saves are generally read and written by the cores directly, we just tell it a root path to use for file placement.

i think we can just make a PR for this as-is. please do that and update the metadata and ticket relationships on tis ticket to any new tickets or PRs created. Thank youl
